edgelet/edgelet-http-external-provisioning/src/client/external_provisioning.rs

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::sync::Arc;
+
+use external_provisioning::apis::client::APIClient;
+use external_provisioning::apis::configuration::Configuration;
+use external_provisioning::apis::ExternalProvisioningApi;
+use external_provisioning::models::*;
+use failure::{Fail, ResultExt};
+use futures::prelude::*;
+use hyper::Client;
+use url::Url;
+
+use edgelet_core::UrlExt;
+use edgelet_http::UrlConnector;
+
+use crate::error::{Error, ErrorKind};
+
+pub trait ExternalProvisioningInterface {
```

Comment:
`ExternalProvisioningInformation` perhaps. I don't like the `Interface` name.

Or could it be used for more things that just "get device provisioning information" in the future?

Reply:
Yes, it can potentially be used for more things. I can rename it to just ExternalProvisioning. How does that sound?

Reply:
I didn't suggest `ExternalProvisioning` because you also have the `provisioning::ExternalProvisioning<T, U>` struct with `T: ExternalProvisioningInterface` right now, so it would look a bit silly to name them both the same.

Reply:
You can leave it as it is if there isn't a better name.

Reply:
Yeah, can't think of any good ones. Gonna leave it as is then